tealium/collectors/momentsapi/MomentsAPIExtensions.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  MomentsAPIExtensions.swift
+//  tealium-swift
+//
+//  Copyright © 2024 Tealium, Inc. All rights reserved.
+//
+
+import Foundation
+#if momentsapi
+import TealiumCore
+#endif
+
+extension TealiumConfigKey {
+    static let momentsAPIRegion = "moments_api_region"
+    static let momentsAPIReferer = "moments_api_referer"
+}
+
+public enum MomentsAPIRegion: String {
+    // swiftlint:disable identifier_name
+    case germany = "eu-central-1"
+    case us_east = "us-east-1"
+    case sydney = "ap-southeast-2"
+    case oregon = "us-west-2"
+    case tokyo = "ap-northeast-1"
+    case hong_kong = "ap-east-1"
+    // swiftlint:enable identifier_name
+}
+
+public extension TealiumConfig {
+
+    /// Sets the region for calls to the Moments API endpoint
+    var momentsAPIRegion: MomentsAPIRegion? {
+        get {
+            options[TealiumConfigKey.momentsAPIRegion] as? MomentsAPIRegion
+        }
+
+        set {
+            options[TealiumConfigKey.momentsAPIRegion] = newValue
+        }
+    }
+
+    /// Sets the region for calls to the Moments API endpoint
+    var momentsAPIReferer: String? {
+        get {
+            options[TealiumConfigKey.momentsAPIReferer] as? String
+        }
+
+        set {
+            options[TealiumConfigKey.momentsAPIReferer] = newValue
+        }
+    }
+}
+
+public extension Tealium {
+
+    class MomentsAPIWrapper {
+        private unowned var tealium: Tealium
```

Comment:
Maybe `weak` instead of `unowned` here is safer.

I can see this causing a crash:
```swift
var tealium: Tealium? = Tealium(..)
let momentsAPI = tealium?.momentsAPI
tealium = nil
momentsAPI?.fetchEngineResponse(...) // Crash!
```